arcade/arcade/core/catalog.py

```diff
@@ -481,10 +482,10 @@ def create_output_definition(func: Callable) -> ToolOutput:
         return_type = return_type.__origin__
 
     # Unwrap Optional types
-    is_optional = False
-    if get_origin(return_type) is Union and type(None) in get_args(return_type):
+    # Both Optional[T] and T | None are supported
+    is_optional = is_strict_optional(return_type)
+    if is_optional:
```

Comment:
nit: could be combined
```suggestion
    if is_strict_optional(return_type):
```

Reply:
saving it in variable `is_optional` so that I don't have to call `is_strict_optional` again on line 493